src/utils/bytes.js

```diff
@@ -38,6 +37,21 @@ export function utf8Decode(bytes) {
   });
 }
 
+// TODO(aghassemi, #6139): Remove the async version of utf8 encoding and rename
+// the sync versions to the canonical utf8Decode/utf8Encode.
+/**
+ * Interpret a byte array as a UTF-8 string.
+ * @param {!BufferSource} bytes
+ * @return {!string}
+ */
+export function utf8DecodeSync(bytes) {
+  if (typeof TextDecoder !== 'undefined') {
+    return new TextDecoder('utf-8').decode(bytes);
+  }
+  const asciiString = bytesToString(new Uint8Array(bytes));
```

Comment:
We need to construct the new typed array from the `bytes`'s `buffer`, else they get down casted:

```js
var u16 = new Uint16Array([42434]);
var u = new Uint8Array(u16);
// => [194], should be [194, 165]

new TextDecoder('utf-8').decode(u16);
// => "¥"
new TextDecoder('utf-8').decode(u);
// => "�"
```

Since we're allowing `ArrayBuffer` as an input, `new Uint8Array(bytes.buffer || bytes)`